libs/Smarty_Compiler.class.php

```diff
@@ -1955,7 +1955,11 @@ function _parse_modifiers(&$output, $modifier_string)
                 $_modifier_args = '';
 
             if ($_map_array) {
-                $output = "((is_array(\$_tmp=$output)) ? \$this->_run_mod_handler('$_modifier_name', true, \$_tmp$_modifier_args) : " . $this->_compile_plugin_call('modifier', $_modifier_name) . "(\$_tmp$_modifier_args))";
+                if(strtolower($_modifier_name) == 'count'){
+                    $output = "((is_array(\$_tmp=$output)) ? \$this->_run_mod_handler('$_modifier_name', true, \$_tmp$_modifier_args) : ((is_array(\$_tmp) || \$_tmp instanceof Countable)?" .  $this->_compile_plugin_call('modifier', $_modifier_name)."(\$_tmp$_modifier_args):(is_null(\$_tmp)?0:1)))";
```

Comment:
Won't `is_array(\$_tmp) || ` always be false here?

Reply:
Not sure if i understand your question. So the idea here is just that i make a check for the $_tmp to be either array or instance of countable. if this is the case we call normally, but if both of them are false we a second check is_null(\$_tmp)?0:1
so that we return the number 0 if $_tmp is null and otherwise its 1.

Reply:
and sorry for the late response. you've been on my list and corona unfortunatly made us go into "war mode" at the company ;)

Reply:
Sorry didnt see i had to click resolve conversation (havent made any pull requests before)